src/powerset.rs

```diff
@@ -81,6 +81,12 @@ impl<I> Iterator for Powerset<I>
             (0, self_total.1)
         }
     }
+
+    fn count(self) -> usize {
+        let k = self.combs.k();
+        let (n, combs_count) = self.combs.n_and_count();
+        combs_count + (k + 1..=n).map(|i| checked_binomial(n, i).unwrap()).sum::<usize>()
```

Comment:
`reset(i)` starts by setting `first` to true, then `remaining_for(n, **true**, indices)` returns `checked_binomial(n, i)` (unless `n < i` but it won't happen because the range `k+1..=n` ensures us that `i <= n`).

And as you said, it's consumed. I already had to create the "weird" `n_and_count` to get `n` a second time (the first time being to compute `combs.count`).
And the test function tests counts one by one, it seems tested well enough to me.

The only improvement I can think of here is `if k == 0 { return 2.pow(n) - 1 + combs_count; }` and yet this could overflow in a very specific case (`n == usize::BITS`) while my sum would not necessary overflow (depends on `combs_count`).